src/components/QueryRunner.ts

```diff
@@ -0,0 +1,25 @@
+import * as vscode from 'vscode'
+
+import { Store } from './Store'
+import { IInstance } from '../types'
+import { TableView } from '../views/TableView'
+import { QueryResult } from '../models'
+import { APIClient } from './APIClient'
+
+export async function runQuery(query : string, context : vscode.ExtensionContext) : Promise<void> {
```

Comment:
Can you add a doc here? Main looking to understand the context of when I should use this method. Based on the implementation it seems that its the highest level abstraction to run a query from within the plugin. Meaning it does everything needed to run a query without any other context. For example it creates the client gets the creds from the store etc. Is that a correct understanding?